frontend/storage/reducers/gameSpoilersReducer.ts

```diff
@@ -0,0 +1,26 @@
+// eslint-disable-next-line no-unused-vars
+import {AppStore, SET_GAME_SPOILERS, AppActions} from '../common';
+// eslint-disable-next-line no-unused-vars
+import { GameSpoilers, GameSpoilersMutator } from '../models/GameSpoilers';
+
+export function gameSpoilersReducer(state: AppStore, action: AppActions): AppStore {
+	switch (action.type) {
+		case SET_GAME_SPOILERS: {
+			return mutateGameSpoilers(state, GameSpoilersMutator.changeShowSpoilers(
```

Comment:
Once you replace the game spoilers with a map of `gameId` => `boolean` you'll have to remember that in Redux you're generally expected to NOT modify the state, but rather create clones that have the changes applied. So you'll have to first clone the old spoiler list, apply the change to the clone and put the clone into the state.